salt/beacons/inotify.py

```diff
@@ -105,37 +106,42 @@ def __validate__(config):
     ]
 
     # Configuration for inotify beacon should be a dict of dicts
-    log.debug('config {0}'.format(config))
-    if not isinstance(config, dict):
-        return False, 'Configuration for inotify beacon must be a dictionary.'
+    if not isinstance(config, list):
+        return False, 'Configuration for inotify beacon must be a list.'
     else:
-        for config_item in config:
-            if not isinstance(config[config_item], dict):
+        _config = {}
+        list(map(_config.update, config))
+
+        for path in _config.get('files', {}):
+
+            if not isinstance(_config['files'][path], dict):
```

Comment:
The outer part of the beacon configuration is still a list but for some of them, they're still using dictionaries, similar to how the configuration for many state modules is.